app/api/roles.py

```diff
@@ -74,3 +80,45 @@ def before_delete_object(self, obj, kwargs):
                   'methods': {
                       'before_get_object': before_get_object
                   }}
+
+
+@role_misc_routes.route('/change-organiser', methods=['POST'])
+def change_organiser():
+    current_organiser_id = request.json['data']['current_org_id']
+    new_organiser_id = request.json['data']['new_org_id']
+    event_id = request.json['data']['event_id']
+
+    # Check if the user is actually an organiser of the event
+    # Delete the User Event Role
+    try:
+        role_entry_organiser_to_delete = UER.query.filter_by(user_id=current_organiser_id,
+                                                             event_id=event_id,
+                                                             role_id=1).one()
+        db.session.delete(role_entry_organiser_to_delete)
+
+    except NoResultFound:
+        return NotFoundError({'source': ''}, 'Organiser Role Entry not found').respond()
+
+    except Exception as e:
+        return ServerError({'source': ''}, e).respond()
+
+    # Check if the coorganiser is not found but the earlier entry has been deleted
+    # Rollback the last commit in case of exception to restore organiser status
+    try:
+        role_entry_coorg_to_organiser = UER.query.filter_by(user_id=new_organiser_id,
+                                                            event_id=event_id,
+                                                            role_id=2).one()
+        role_entry_coorg_to_organiser.role_id = 1
+        save_to_db(role_entry_coorg_to_organiser)
+
+    except NoResultFound:
+        db.session.rollback()
+        return NotFoundError({'source': ''}, 'Coorganiser Role Entry not found').respond()
+
+    # Handle Unexpected Errors
+    except Exception as e:
+        return ServerError({'source': ''}, e).respond()
+
+    return jsonify({
```

Comment:
@shreyanshdwivedi Could you suggest how would a successful transaction be handled on the frontend?
Also, we need to open a new issue on front end to implement an option to transfer organiser

Reply:
Okay. So for frontend what we'll do is we'll send a update request to change organizer email, which will retrieve user ID of new organizer and update the db and send him/her an email regarding the change to both the new and old organizer. We'll have email messages too.
We'll have to discuss this extensively.